From 571a47f44bb8fa998f41c3b07d74bf79f4146116 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sat, 15 Oct 2022 09:10:05 +0300
Subject: [PATCH] [SW-0] gaudi3: initialize boot desc flag

boot_desc_flag for SPI and EROM manifest image is stored in
non persist ROM structure and is not initialized as part of ROM init.
That means other FW components might use the DCCM section which can
alter boot_desc_flag if used in ROM without initialization. The purpose
of boot_desc_flag is only to prevent re-initialization of boot desc
for given boot and does not need to retain value for next boot (ie.
reset). This patch initialize boot_desc_flag to 0 as part of ROM init
routine.

Change-Id: Ia2bebff55673ef1ce06740be968d9cdc7c90cdc8
---

diff --git a/arch/gaudi3/arc/common.c b/arch/gaudi3/arc/common.c
index 093aa4d..241be71 100644
--- a/arch/gaudi3/arc/common.c
+++ b/arch/gaudi3/arc/common.c
@@ -440,6 +440,7 @@
 								HL_BOOT_PTR_EROM_DMA_DEST;
 	arcboot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_EROM].dest_addr_arc =
 								HL_BOOT_PTR_EROM_DMA_DEST_ARC;
+	arcboot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_EROM].boot_desc_flag = 0;
 
 	arcboot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].img_ptr_spi_offset =
 								HL_SPI_IMG_PTRS_BASE;
@@ -447,6 +448,7 @@
 								HL_BOOT_PTR_SPI_DMA_DEST;
 	arcboot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].dest_addr_arc =
 								HL_BOOT_PTR_SPI_DMA_DEST_ARC;
+	arcboot_rom_manifest.boot_ptr_ctx[HL_FW_BOOT_PTR_SPI].boot_desc_flag = 0;
 }
 
 #define HL_ARC1_DCCM_PERSIST_SECTION_SIZE			(3 * 1024) /* 3KB */
